RxSwift/Units/Single+Maybe+Completeable.swift

```diff
@@ -0,0 +1,455 @@
+//
+//  Single+Maybe+Completeable.swift
+//  RxSwift
+//
+//  Created by Krunoslav Zaher on 3/5/17.
+//  Copyright © 2017 Krunoslav Zaher. All rights reserved.
+//
+
+/// Common functionality of all a push style sequences containing 0 or 1 element.
+public struct PrimitiveSequence<Proof, Element> {
+    fileprivate let source: Observable<Element>
+
+    fileprivate init(raw: Observable<Element>) {
+        self.source = raw
+    }
+}
+
+public enum SingleProof { }
```

Comment:
I'm not sure what the best naming would be here. 

The purpose of that type was to use generic parameter `PrimitiveSequence` to mark that certain properties where proved.

We've used `SharingStrategy` on `SharedSequence` because it's purpose is to enable customization of sharing strategy.

Using proof was also little weird to me. I would like to find some better name.